packages/loader/container-loader/src/utils.ts

```diff
@@ -33,15 +33,20 @@ export interface IParsedUrl {
 }
 
 export function parseUrl(url: string): IParsedUrl | undefined {
-    const parsed = parse(url, true);
+    const parsed = new URL(url);
     if (typeof parsed.pathname !== "string") {
         throw new Error("Failed to parse pathname");
     }
-    const query = parsed.search ?? "";
+
+    // TODO: this is inelegant... is there a better way?
+    let queryParamCount = 0;
+    parsed.searchParams.forEach(() => queryParamCount++);
```

Comment:
I have to be missing something, but I can't figure out a better way to do this... Help! :)

Reply:
I only need to know if the URL had query params, not the actual count. So, some options include:

If the searchParams property is undefined in the case where the input URL had no query string, then I could just check if the property is undefined. But according to the docs and brief local testing, the searchParams object is defined regardless of whether the URL had a query string. So that option's out.

Since I only need to know if there is at least one item, I could restructure it as a for...of loop and short-circuit the loop after one iteration.

```ts
let hasParams = false;
for(const _ of parsed.searchParams.keys()) {
    hasParams = true;
    break;
});

Reply:
I was being dumb. In the case of no query string, the `search` property will return the empty string. So I can just use the original code. <sigh>